src/services/azureFunctionsService.ts

```diff
@@ -94,9 +95,17 @@ export class AzureFunctionsService implements mssql.IAzureFunctionsService {
 			folderPath: projectFile
 		});
 
+		// check for the new function file to be created
+		const functionFile = await newFilePromise;
+		const timeoutFunctionFile = await checkFunctionFile;
+		try {
+			await Promise.race([functionFile, timeoutFunctionFile]);
```

Comment:
This isn't how you use Promise.race - you shouldn't be awaiting the promises above. It should look like this:

```
const timeoutPromise = waitForPromise();
const functionFile = await Promise.race([newFilePromise, timeoutPromise]);
```

But now there's the slight problem since the type of functionFile is going to be unknown (since you have mixed promise types). There's a couple solutions : 

1. Just cast the return type to string since you know that if it resolves it'll always be a string
2. Have the timeout promise be generic with the same type as the other promise (string in this case)
3. Use a different way of doing the timeout, for example [here](https://github.com/microsoft/azuredatastudio/blob/2bc6a0cd012de69504061a876070cfa43533fedc/src/vs/base/common/async.ts#L79) they're just having it return undefined if it times out. So you could use that functionality and then check if it's undefined after the await and throw in that case.